### PR TITLE
change behavior of help command

### LINKED ISSUE #939: help: prefer file over label
On our recent Fedora activity day, we decided that we would prefer atomic command to pick file in a container over label. Right now it's the other way around, label is preferred over file.

Would you be interested in such changes? If so, I will send a PR.

https://pagure.io/atomic-wg/issue/256

### PR DESCRIPTION
**Pull request by TomasTomecek**
Fixes #939

Reference: https://pagure.io/atomic-wg/issue/256#comment-436537

 * The command now prefers files over "help" label.
   * Help files can be one of:
     * help.1, README.md

### TODO
 * [x] code
 * [x] tests
 * [x] documentation

Can I get a code review please?

Signed-off-by: Tomas Tomecek <ttomecek@redhat.com>

### CODE REVIEW

**Comment by TomasTomecek** [comment]
Those pylint warnings are easy to fix: how do I run the integration test suite?

**Comment by yuqi-zhang** [comment]
You can run the test suite locally with `sudo make test`, or alternatively you can run a single test like: `sudo TEST_INTEGRATION=diff sh test.sh`

To run the integration tests you have to pass the pylint errors, so if you want to re-trigger the tests here, just submit a fixup to the pylint issues and the bot will re-run the tests.

**Comment by TomasTomecek** [comment]
@yuqi-zhang thank you! Looks like it's running now.

File: Atomic/mount.py [hunk 1]
```diff
@@ -732,6 +732,28 @@ def _clean_temp_container_by_path(self, path):
             self.d.remove_container(short_cid)
         self._clean_tmp_image()
 
+
+class DockerMountContextManager(object):
```
Comment (baude) [inline Atomic/mount.py]: I *really* love this piece of code.  It could be used quite a bit to replace less efficient code in atomic.  A couple of thoughts:

* Can you write one or more unit tests for it and maybe one integration test?
* Is there any way we could modify this so that if an exception occurred, it would clean up after itself?

Comment (TomasTomecek) [inline Atomic/mount.py]: These are great suggestions! Maybe I could move that code into separate PR. It will likely take me a couple days.

**Comment by dustymabe** [comment]
bump

**Comment by rhatdan** [comment]
I would like to create a new version of atomic, any chance we can move this pull request along.

**Comment by yuqi-zhang** [comment]
I've tested the functionality and it looks good. If you're ok with the implementation @baude I can help port this over to system containers after this PR gets merged.

**Comment by TomasTomecek** [comment]
Sorry for lack of response, but I was really busy lately. I am planning to address this pull request (and others) this week.

**Comment by TomasTomecek** [comment]
Managed to split the `DockerMountContextManager` into #976. Will rebase this onto #976 in the meanwhile to finish this PR.

**Comment by TomasTomecek** [comment]
Please don't merge this before #976.

This should be ready for review now. Let me know whether I should write some tests.

**Comment by baude** [comment]
@TomasTomecek Please do include tests. We really want to be able to protect against regression.

**Comment by TomasTomecek** [comment]
Will progress with this one once #976 is merged so I don't have to rebase/update this one constantly.

**Comment by TomasTomecek** [comment]
I have finally had time to take care of this PR: I rebased, updated code, provided more tests.

**Comment by TomasTomecek** [comment]
I just added manpage, so this PR is ready for complete review.

With these changes atomic should comply to Fedora container guidelines (cc @jberkus @dustymabe).

**Comment by TomasTomecek** [comment]
I am struggling with CI: the test works locally, but fails in CI :/

**Comment by yuqi-zhang** [comment]
I think the "atomic-test-2" image has been gone for awhile, is there a reason why its being used in the test?

**Comment by TomasTomecek** [comment]
@yuqi-zhang thanks for headsup: so it looks like that I have that image available locally from the past

**Comment by TomasTomecek** [comment]
The CI failure is caused by some other test installing centos to ostree storage hence atomic can't find the image in docker storage.

File: tests/integration/test_help.sh [hunk 2]
```diff
@@ -16,27 +18,46 @@ IFS=$'\n\t'
 
 OUTPUT=$(/bin/true)
 
+# let's clean the environment first
+${ATOMIC} uninstall -f --storage=docker centos || true
+${ATOMIC} uninstall -f --storage=ostree centos || true
```
Comment (yuqi-zhang) [inline tests/integration/test_help.sh]: Are you trying to delete images? Uninstall only handles "uninstall" labels for docker containers and removes the container for system containers, which shouldn't affect this test at all. I think what you want is "atomic images delete"?

Comment (TomasTomecek) [inline tests/integration/test_help.sh]: I wasn't able to reproduce either, that's why I put these uninstall commands in the test. Thanks for `atomic images delete`, trying that now.

**Comment by TomasTomecek** [comment]
@yuqi-zhang thank you so much for the `$ ostree` command suggestion, tests are passing now, @baude ready for review.

File: docs/atomic-help.1.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+% May 2017
+# NAME
+atomic-help - Get help for container images
+
+# SYNOPSIS
+**atomic help**
+[**-h**|**--help**]
+IMAGE
+
+**atomic help** provides documentation for specified container IMAGE.
```
Comment (TomSweeneyRedHat) [inline docs/atomic-help.1.md]: nit 'for specified' to 'for the specified'

Comment (TomasTomecek) [inline docs/atomic-help.1.md]: thank you! since I'm not native, it's hard for me to get documentation right

**Comment by TomasTomecek** [comment]
updated

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ 097fc88

**Comment by TomasTomecek** [comment]
\o/
